core/templates/dev/head/i18n.js

```diff
@@ -13,34 +13,81 @@
+    'I18N_GALLERY_PAGE_TITLE': 'Gallery',
+    'I18N_GALLERY_PAGE_SUBTITLE': 'Oppia',
+    'I18N_GALLERY_LOADING': 'Loading',
+    'I18N_SIGNUP_PAGE_SUBTITLE': 'Registration',
+    'I18N_SIGNUP_PAGE_TITLE': 'Oppia',
+    'I18N_GALLERY_SEARCH_PLACEHOLDER': 'What are you curious about?',
+    'I18N_GALLERY_ALL_LANGUAGES': 'All Languages',
+    'I18N_GALLERY_ALL_CATEGORIES': 'All Categories',
+    'I18N_GALLERY_CREATE_EXPLORATION': 'Create exploration',
+    'I18N_SIDEBAR_HOME_LINK': 'Home',
+    'I18N_SIDEBAR_HOME_ABOUT': 'About',
+    'I18N_SIDEBAR_PARTICIPATION_PLAYBOOK': 'Participation Playbook',
+    'I18N_SIDEBAR_FORUM': 'Forum',
+    'I18N_SIDEBAR_FOLLOW_US': 'Follow Us',
+    'I18N_SIDEBAR_ADDITIONAL_LINK_SITE_FEEDBACK': 'Site Feedback',
+    'I18N_TOPNAV_SIGN_IN': 'Sign in',
+    'I18N_SIGNUP_REGISTRATION': 'Registration',
+    'I18N_SIGNUP_LOADING': 'Loading'
+};
+
 oppia.constant('SUPPORTED_LANGUAGES', {
   'en': 'English',
   'es': 'Español'
 });
 
 oppia.controller('I18nFooter', [
-    '$scope', '$translate', 'SUPPORTED_LANGUAGES',
-    function($scope, $translate, SUPPORTED_LANGUAGES) {
+    '$rootScope', '$scope', '$translate', 'SUPPORTED_LANGUAGES',
+    function($rootScope, $scope, $translate, SUPPORTED_LANGUAGES) {
   $scope.SUPPORTED_LANGUAGES = SUPPORTED_LANGUAGES;
   // Changes the language of the translations.
-  $scope.changeLanguage = function (langCode) {
+  $scope.changeLanguage = function(langCode) {
     $translate.use(langCode);
   };
+  // After loading default translations, change the language for the storaged
+  // language if necessary
+  $rootScope.$on('$translateLoadingSuccess', function (event, args) {
+    var currentLang = $translate.proposedLanguage() || $translate.use();
+    $translate.use(currentLang);
+  });
 }]);
 
 oppia.config([
-    '$translateProvider', '$translatePartialLoaderProvider',
-    function($translateProvider, $translatePartialLoaderProvider) {
+    '$translateProvider', 'SUPPORTED_LANGUAGES',
+    function($translateProvider, SUPPORTED_LANGUAGES) {
+  var availableLanguageKeys = [];
+  var availableLanguageKeysMap = {};
+  for (var prop in SUPPORTED_LANGUAGES) {
+    availableLanguageKeys.push(prop);
+    availableLanguageKeysMap[prop + '*'] = prop;
+  };
+  availableLanguageKeysMap['*'] = 'en';
+  $translateProvider.registerAvailableLanguageKeys(
+    availableLanguageKeys, availableLanguageKeysMap);
+
+  $translateProvider.translations('en', defaultTranslations);
+  $translateProvider.fallbackLanguage('en');
+
+  $translateProvider.determinePreferredLanguage();
+  $translateProvider.useCookieStorage();
+
   $translateProvider.useLoader('$translatePartialLoader', {
     urlTemplate: '/i18n/{part}/{lang}.json'
   });
-  $translatePartialLoaderProvider.addPart('sidenav');
-  $translatePartialLoaderProvider.addPart('topnav');
-  $translateProvider.preferredLanguage('en');
-  $translateProvider.fallbackLanguage('en');
+  // using strategy 'sanitize' does not support utf-8 encoding.
```

Comment:
Yes, it is closed because it has a solution: not to use sanitize... at least they let you know is not going to work!